archive/proc.go

```diff
@@ -12,6 +14,82 @@ import (
 
 var ErrSyntax = errors.New("syntax/format error encountered parsing zng data")
 
+// A FieldCutter is a custom proc that, given an input record and a
+// field name, outputs one record per input record containing the
+// field, with the field as only column.  It is used for field-based
+// indexing.
+type FieldCutter struct {
```

Comment:
Was it impractical to alter the cut proc to add the type tracking?

Would you add to the comment here, reflecting your commit comment, that this was put in because the cut proc couldn't track type changes?

Reply:
I don't think it would have been impractical, but I didn't want to add a "error on type changes" mode to the cut proc as it didn't seem to belong there and wouldn't be exposed from zql. And FieldCutter is way simpler/shorter than cut proc (the cut proc actually does a whole bunch of work to enable cutting of different types).